utils/utils.go

```diff
@@ -41,6 +47,33 @@ func CheckStakeWeightExceedsThreshold(accumulatedSignatureWeight *big.Int, total
 	return scaledTotalWeight.Cmp(scaledSigWeight) != 1
 }
 
+//
+// Chain Utils
+//
+
+// Calls f until it returns a non-error result or the context is canceled, with a 200ms delay between calls.
+func CallWithRetry[T any](ctx context.Context, f func() (T, error)) (T, error) {
+	queryTicker := time.NewTicker(200 * time.Millisecond)
+	defer queryTicker.Stop()
+	var (
+		t   T
+		err error
+	)
+	for {
+		t, err = f()
```

Comment:
Should we state in the comments that we assume the time to call f() is significant less than the ticker delay? This should generally be true unless f() itself blocks or has some sleep logic. 

Otherwise the ticker will not have the intended effect, and we should only start the ticker after f() completes in each loop iteration.

Reply:
Go's `time.Ticker` type "will adjust the time interval or drop ticks to make up for slow receivers" ([docs](https://pkg.go.dev/time#NewTicker)). This is implemented using a buffered channel with a single element; if the client falls behind, new ticks are dropped.

I don't think we should make any assumptions on `f`, especially since `f` is usually a blocking network request. `time.Ticker` should account for arbitrarily expensive `f`s though.